### PR TITLE
adds unit testing

### PR DESCRIPTION
**Pull request by vmcvlad**

### CODE REVIEW

File: tests/features/DiscussionTest.php [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        //$this->withoutExceptionHandling();
+
+        $this->seed()
+            ->actingAs(User::first());
+
+        $this->testModel = $this->model();
+    }
+
+    /** @test */
+    public function can_get_discussions_index()
```
Comment (aocneanu) [inline tests/features/DiscussionTest.php:39]: first create a discussion using the factory, then check that in the index the created discussion is present

File: tests/features/DiscussionTest.php [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        ], false))->assertStatus(200);
+    }
+
+    /** @test */
+    public function can_store_discussion()
+    {
+        $discussion = factory(Discussion::class)->make([
+            'discussable_id' => $this->testModel->id,
+            'discussable_type' => DiscussionTestModel::class,
+        ]);
```
Comment (aocneanu) [inline tests/features/DiscussionTest.php]: move this to the setup method to DRY the code

File: tests/features/DiscussionTest.php [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    }
+
+    /** @test */
+    public function can_update_discussion()
+    {
+        $discussion = factory(Discussion::class)->create([
+            'discussable_id' => $this->testModel->id,
+            'discussable_type' => DiscussionTestModel::class,
+        ]);
+
```
Comment (aocneanu) [inline tests/features/DiscussionTest.php]: the same

File: tests/features/DiscussionTest.php [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    /** @test */
+    public function can_only_act_on_allowed_discussion()
+    {
+        $discussion = factory(Discussion::class)->create([
+            'discussable_id' => $this->testModel->id,
+            'discussable_type' => DiscussionTestModel::class,
+        ]);
+
+        $this->actingAs($this->createForeignUser());
```
Comment (aocneanu) [inline tests/features/DiscussionTest.php]: rename `createForeignUser` to `anotherUser`

File: tests/features/DiscussionTest.php [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+            $discussion->toArray() + [
+                'path' => $this->faker->url,
+            ]
+        )->assertStatus(403);
+    }
+
+    /** @test */
+    public function can_store_reply_to_discussion()
+    {
+        $reply = Reply::make([
```
Comment (aocneanu) [inline tests/features/DiscussionTest.php]: create a reply factory

Comment (aocneanu) [reply tests/features/DiscussionTest.php]: move all the reply logic to a dedicated test

File: tests/features/DiscussionTest.php [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+            'reactableId' => factory(Discussion::class)->create([
+                    'discussable_id' => $this->testModel->id,
+                    'discussable_type' => DiscussionTestModel::class,
+                ])->id,
+            'reactableType' => 'discussion',
+            'type' => Reactions::Clap,
+            'userId' => auth()->user()->id,
+        ]);
+
+        $this->post(route('core.discussions.react'), $reactionParams->toArray())->assertStatus(200);
```
Comment (aocneanu) [inline tests/features/DiscussionTest.php]: don't make the `reactParam` a collection only to convert it toArray

File: tests/features/DiscussionTest.php [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+    }
+
+    private function model()
+    {
+        $this->createTestTable();
+
+        return DiscussionTestModel::create(['name' => 'discussable']);
```
Comment (aocneanu) [inline tests/features/DiscussionTest.php]: let's move this method in the setup method. You would not expect a `model()` method to create a model and something else

File: tests/features/DiscussionTest.php [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+            $table->timestamps();
+        });
+
+        return $this;
+    }
+
+    private function createForeignUser()
+    {
+        return factory(User::class)->create([
+            'role_id' => function () {
```
Comment (aocneanu) [inline tests/features/DiscussionTest.php]: why is this needed? doesn't the user factory do this by itself?

File: tests/features/DiscussionTest.php [hunk 9]
```diff
@@ -28,206 +24,108 @@ protected function setUp()
...
     {
-        $discussion = factory(Discussion::class)->create([
-            'discussable_id' => $this->testModel->id,
-            'discussable_type' => DiscussionTestModel::class,
-        ]);
-
-        $this->delete(route('core.discussions.destroy', $discussion->id, false))
+        $this->delete(route('core.discussions.destroy', $this->testModel->id, false))
             ->assertStatus(200);
     }
```
Comment (aocneanu) [inline tests/features/DiscussionTest.php:78]: here we need an additional query to check that the model exists in the db

File: tests/features/DiscussionTest.php [hunk 10]
```diff
@@ -28,206 +24,108 @@ protected function setUp()
...
-                    'discussable_id' => $this->testModel->id,
-                    'discussable_type' => DiscussionTestModel::class,
+        return factory(Discussion::class)->make([
+            'discussable_id' => DiscussionTestModel::create([
+                    'name' => 'discussable'
                 ])->id,
-            'reactableType' => 'discussion',
-            'type' => Reactions::Clap,
-            'userId' => auth()->user()->id,
+            'discussable_type' => DiscussionTestModel::class,
```
Comment (aocneanu) [inline tests/features/DiscussionTest.php:115]: let's dump this method and move its content in setUp()

File: tests/features/ReplyTest.php [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+            'discussion_id' => factory(Discussion::class)->create()->id
+        ]);
+    }
+
+    private function postParams()
+    {
+        return factory(Reply::class)->make([
+            'discussion_id' => factory(Discussion::class)->create()->id
+        ]);
+    }
```
Comment (aocneanu) [inline tests/features/ReplyTest.php]: moved in setUp()

File: tests/features/DiscussionTest.php (hunk 1 above)
Comment (vmcvlad) [reply tests/features/DiscussionTest.php:39]: addressed in cc9077a
